libs/application-generic/src/services/resource-validator.service.ts

```diff
@@ -63,36 +82,31 @@ export class ResourceValidatorService {
   }
 
   private async getWorkflowLimit(environment: EnvironmentEntity, organization: OrganizationEntity) {
-    const maxWorkflowSystemLimit = await this.getMaxWorkflowSystemLimit(environment, organization);
+    const systemLimitMaxWorkflow = await this.getSystemLimitWorkflows(environment, organization);
+
+    // If the system limit is not the default, we need to use it as the absolute limit for special cases instead of the tier limit
+    const isSpecialLimit = systemLimitMaxWorkflow !== SYSTEM_LIMITS.WORKFLOWS;
+    if (isSpecialLimit) {
+      return systemLimitMaxWorkflow;
+    }
+
     const maxWorkflowsTierLimit = await this.getMaxWorkflowsTierLimit(environment, organization);
 
-    return Math.min(maxWorkflowSystemLimit, maxWorkflowsTierLimit);
+    return Math.min(systemLimitMaxWorkflow, maxWorkflowsTierLimit);
```

Comment:
i wonder if we should create a reusable function and reuse this logic in the tier validation usecase as well. 
For now, it felt like overkill, so I decided to postpone it until we have another use case for this "new era" ff logic.

Reply:
I agree, it's an overkill for now.